packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -43,6 +43,11 @@ export class ObjectForest extends SimpleDependee implements IEditableForest {
     }
 
     public root(range: DetachedField, index = 0): Anchor {
+        if (!this.roots.get(range)?.length) {
+            return this.anchors.track(null);
```

Comment:
Perhaps this API should return 'Anchor | undefined' and the notion of a special 'NeverAnchor' can be removed?

Reply:
If we use `undefined` then the caller has to handle it. Seeing that they already need to check `NavigationResult` anyway (e.g. see getTreeString() in forestIndex.ts), this seemed more elegant (for the caller) to me. 

The reason for `track` accepting a null argument was Craig's input; he thinks we should leave the door open to add ref-counting of anchors at a later date. By that, I mean keeping track of which anchors are owned and not yet disposed so that we can detect memory leaks (this would be done inside of `track()`). He thought this should apply even to the NeverAnchor, because, even though it is a singleton and undisposed references to it won't ever "leak", we can still enforce the good practice of "always dispose your anchor when you're done with it, regardless of whether it is a special singleton anchor or not". If we don't care about that for the NeverAnchor, because it can't leak, then we could just let this function return the singleton instance directly as you suggested. I am starting to be indifferent, and lean towards just having the singleton be exempt from any ref-counting since it will make the code simpler. @CraigMacomber, are there points I missed?

Reply:
After thinking some more, I realize that making the singleton public wouldn't simplify the code that processes the NeverAnchor, it would only slightly simplify the `track` function. So, I'm going to keep this as it is for now, I think it pollutes the API very little but keeps the door open for the reference-count validation that Craig would eventually like.